### PR TITLE
TCK updates for Servlet 6.0

### PR DESCRIPTION
**Pull request by markt-asf**
---
name: Pull Request
about: Create a pull request for a Platform TCK change
title: 'Servlet 6.0 TCK updates'
labels: ''
assignees: ''

---

**Fixes Issue**
None

**Related Issue(s)**
None

**Describe the change**
Updates required to the Servlet TCK for the Servlet 6.0 release. Includes some commits that need to be back-ported to earlier TCK versions.

**Additional context**
None

CC @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @scottmarlow

### CODE REVIEW

**Comment by markt-asf** [comment]
The "Ensure passed flag is correctly set on test failure" needs to be back-ported.

**Comment by scottmarlow** [comment]
Thanks Mark for working on this, very much appreciated!

**Comment by markt-asf** [comment]
This PR is now at the point where a compliant Servlet 6.0 implementation should pass it. What is missing is new tests for new features.

**Comment by markt-asf** [comment]
Just a rebase. No functional changes.

**Comment by gurunrao** [comment]
@markt-asf  - For readers of commit history, we can squash 17 commits of the PR. Also can we requested Servlet SPEC team to review the PR?

**Comment by markt-asf** [comment]
I'd prefer that the commits weren't squashed into a single commit.

Obviously you have my +1 to these changes. I've asked on the servlet dev list for other committers to comment here.

**Comment by gurunrao** [comment]
> I'd prefer that the commits weren't squashed into a single commit.
> 
> Obviously you have my +1 to these changes. I've asked on the servlet dev list for other committers to comment here.

Please add any volunteers from servlet-dev list as reviewer of the PR.

**Comment by gurunrao** [comment]
> I'd prefer that the commits weren't squashed into a single commit.
> 
> Obviously you have my +1 to these changes. I've asked on the servlet dev list for other committers to comment here.

@markt-asf - Please assign volunteers from servlet-dev for review.

**Comment by markt-asf** [comment]
I asked for additional review volunteers on servlet-dev. No-one stepped forward.

**Comment by pzygielo** [comment]
Should the `HttpUtils` references be removed from ServletJavadocAssertions.html?

**Comment by markt-asf** [comment]
Possibly. Does any actually use the assertion information? I'm wondering if we should continue to maintain it or just remove it entirely.

**Comment by pzygielo** [comment]
Asking only because it was updated in this PR but for other than `HttpUtils` reason.

**Comment by Thihup** [comment]
Is fixing https://github.com/eclipse-ee4j/jakartaee-tck/issues/736 related to this PR?

**Comment by markt-asf** [comment]
@pzygielo If the assertion information stays then references to HttpUtils should be removed but I haven't yet seen an argument for keeping the assertion information.

@Thihup  I haven't noticed any issues related to #736 while working on this but I have been using Java 11.

**Comment by scottmarlow** [comment]
> @Thihup I haven't noticed any issues related to #736 while working on this but I have been using Java 11.

@markt-asf According to https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/src/com/sun/ts/tests/servlet/spec/serverpush/README.md:
```
Once the TCK no longer needs to execute on Java 8, this package can be updated to use the java.net.http module in Java 11 onwards and lib/http.jar may be removed.
```

The servetpush test seems to still work on Java 17 (from WildFly testing with EE 9.1) but from the serverpush/README/md it sounds like lib/http.jar could be removed and the serverpush test changed to use the  java.net.http package since EE 10 will require Java 11+ to run TCKs.  I guess that doesn't have to be tied to this pr for Servlet 6.0 specific changes but it could.

**Comment by markt-asf** [comment]
@scottmarlow Now I remember the hack referenced in #736. If we can get rid of this hack then that would be a good thing. I'll take a look today.

**Review by stuartwdouglas — APPROVED** [review]

File: install/jakartaee/bin/ts.jtx [hunk 1]
```diff
@@ -20,21 +20,6 @@
 # Servlet
 ################
 
-#
```
Comment (scottmarlow) [inline install/jakartaee/bin/ts.jtx:23]: Did we intend to update the copyright date for this file to `2022`?

Comment (scottmarlow) [reply install/jakartaee/bin/ts.jtx:23]: Everything else looks good to me other than ^ question about copyright date.

**Pull request merged**